### ISSUE TITLE
missing renew option if within 24 hours of current certs' expiry

### ISSUE THREAD

**Issue by colourful-land** [open-eid/DigiDoc4-Client#442]
Steps:

1. Start digidoc4
2. Insert Estonian e-residency card.
3. A message pops up "Certificates will expire soon". Clicking that message makes it disappear without any other UI change.

Expected:

3. A message pops up "Your ID card certificates need to be updated" with a "Start" Button.

Environment:

Tested in both Windows 10 and Ubuntu 18.04, the card in question ticks all the checkboxes:

- its certificate is not expired (valid to 18.01.2019 08:59:59 +11:00)
- it was granted before 1 May 2018
- was updated to use the new certificates of ECDSA 384-bit following the security risk announced in the autumn of 2017 before the update window of 31 March 2018.
- it is not in a locked status.

As you can see from the certificate expiration time, this is an urgent matter for me. It will be invalidated real quick. Kindly advise. I can compile a version of digidoc4 if needed.

**Comment by tanelts** [comment]
Thank You for the information. 

To try solve the probleem we would need to see your digidoc4 diagnostics report. 
Instructions for that: [](https://www.id.ee/index.php?id=38889)
Are You using the latest version? [](https://installer.id.ee/?lang=eng)

**Comment by colourful-land** [comment]
Thank you. I have 3 computers. PC / Mac / Ubuntu.

PC: I re-installed digidoc4 by downloading from id.ee at the moment of posting.
Ubuntu: Using DigiDoc 4.2.0.43

**Comment by colourful-land** [comment]
I attempted it on my old mac which has DigiDoc3, which displays "Your ID card does not need updating" while displaying correctly that it expires in less than 24 hours.

**Comment by tanelts** [comment]
Try turning Your computer clock counter-clockwise. For example one day.

**Comment by colourful-land** [comment]
It worked with the 

> Try turning Your computer clock counter-clockwise. For example one day.

It worked. So this support request is turned into a bug report.

When I worked on a mobile app, I thought losing signal right after calling a cab is an edge condition so we didn't design for it. It turns out that this happens frequently because the user would call a cab, wait for it to be done, then enter the elevator to get to the waiting area.

The point I want to make is there is a valid use case that the software must work in "edge condition" like this. My story is that I ordered a Smart Card reader because I don't want to drive 2 hours to meet a friend who has a Smart Card reader. The order didn't show up (turns out it is drop shipped from China), so I had to use plan B, drive 2 hours to borrow the Smart Card reader.

Since so much resource has to be committed to plan B, of course, it is delayed in the last day in the hope plan A would work. I can imagine other users doing the same, so the last day of expiry become the only day they would renew, just like in my story where losing signal only happens after the user called a cab.
